app/views/ForwardLivechatView.tsx

```diff
@@ -18,20 +22,52 @@ const styles = StyleSheet.create({
 	}
 });
 
-const ForwardLivechatView = ({ forwardRoom, navigation, route, theme }) => {
-	const [departments, setDepartments] = useState([]);
-	const [departmentId, setDepartment] = useState();
-	const [users, setUsers] = useState([]);
+// TODO: Refactor when migrate room
+interface IRoom {
+	departmentId?: any;
+	servedBy?: {
+		_id: string;
+	};
+}
+
+interface ITransferData {
+	roomId: string;
+	userId?: string;
+	departmentId?: string;
+}
+
+interface IUser {
+	username: string;
+	_id: string;
+}
+
+interface IParsedData {
+	label: string;
+	value: string;
+}
+
+interface IForwardLivechatViewProps {
+	navigation: StackNavigationProp<any, 'ForwardLivechatView'>;
+	route: RouteProp<{ ForwardLivechatView: { rid: string } }, 'ForwardLivechatView'>;
+	theme: string;
+	forwardRoom: (rid: string, transferData: ITransferData) => void;
+}
+
+const ForwardLivechatView = ({ forwardRoom, navigation, route, theme }: IForwardLivechatViewProps) => {
+	const [departments, setDepartments] = useState<IParsedData[]>([]);
+	const [departmentId, setDepartment] = useState('');
+	const [users, setUsers] = useState<IParsedData[]>([]);
 	const [userId, setUser] = useState();
-	const [room, setRoom] = useState();
+	const [room, setRoom] = useState<IRoom>({});
 
 	const rid = route.params?.rid;
 
 	const getDepartments = async () => {
 		try {
 			const result = await RocketChat.getDepartments();
 			if (result.success) {
-				setDepartments(result.departments.map(department => ({ label: department.name, value: department._id })));
+				const deps = result.departments as ILivechatDepartment[];
```

Comment:
What do you think about this line and the line 86?

Reply:
Why you choose do it like you did as like this?
```const deps: ILivechatDepartment[] = result.departments;```  

Change the line 86 for the same pattern, please :)